server/datastore/mysql/hosts.go

```diff
@@ -942,7 +934,6 @@ func (ds *Datastore) applyHostFilters(opt fleet.HostListOptions, sql string, fil
     %s
```

Comment:
Note: this function is also used by the Count Hosts endpoint, although I haven't (yet) found any use there of the failing policy membership join

Reply:
+1, see https://github.com/fleetdm/fleet/pull/13427#pullrequestreview-1588987970